lib/main_publiccloud.pm

```diff
@@ -233,6 +239,17 @@ sub load_publiccloud_tests {
             load_maintenance_publiccloud_tests();
         } elsif (get_var('PUBLIC_CLOUD_HIMMELBLAU')) {
             loadtest('publiccloud/himmelblau');
+        } elsif (get_var('PUBLIC_CLOUD_APP_IMG')) {
+            my $publiccloud_app_img = get_required_var('PUBLIC_CLOUD_APP_IMG');
+            if (exists($publiccloud_app_img_tests{$publiccloud_app_img})) {
```

Comment:
Such checks are uncommon in the OSADO test framework. We typically would just run the `loadtest` and let it fail if the variables are wrong, since people anyways need to look at the code to see what's wrong.

I leave it to you to decide if you find value in adding this check, I'm just commenting that this is not very common style.